bug(e2e): test_nats_roundtrip fails with engine_unavailable on CI staging

## Summary
The \`e2e\` job in the \`CI\` workflow fails on every run against \`staging\` with:

\`\`\`
AssertionError: ok is not True. full reply:
{'error': 'engine_unavailable', 'contract_version': '1', 'ok': False, 'request_id': '…'}
\`\`\`

Test: \`tests/e2e/test_nats_roundtrip.py::test_nats_roundtrip\`
Stub: \`tests/e2e/stub_hub.py:94\` (\`_assert_reply\`)

The TTS satellite responds to the hub's request with \`ok: false, error: engine_unavailable\` before generating any audio. The envelope is well-formed (contract_version + request_id echo correct) — the failure is on the engine side, not the protocol.

## Evidence this is pre-existing, not a regression

| Run | Branch | Head commit | Conclusion |
|---|---|---|---|
| 24579780924 | staging | #78 (chore(release): align with Convention A, 2026-04-17) | ❌ e2e — identical failure |
| 24613748499 | feat/79-install-quality-gates | quality-gates install | ❌ e2e — identical failure |

Pattern holds for at least the 5 most recent staging runs (\`gh run list --repo Roxabi/voiceCLI --branch staging --limit 10\`).

## Reproduction

\`\`\`bash
gh workflow run CI --ref staging --repo Roxabi/voiceCLI
# wait for e2e job → fails identically
\`\`\`

The \`ci\` job (lint, typecheck, 416 unit tests) passes green alongside. Only \`e2e\` is affected.

## Hypothesis

1. The TTS satellite in the compose stack is coming up but the engine (mock or real) never transitions to ready, so every \`generate\` request short-circuits to \`engine_unavailable\`.
2. Likely a recent change to the satellite startup / ready-signal contract, or a drift between the docker-compose stack definition and what the satellite expects.

## Acceptance

- [ ] Root cause identified (satellite startup, engine loader, compose topology, or test harness)
- [ ] \`tests/e2e/test_nats_roundtrip.py::test_nats_roundtrip\` passes on staging CI
- [ ] Green \`e2e\` job becomes the new baseline so future PRs can rely on it

## Reference

- Discovered while landing #79 (quality-gates install) — PR #84 surfaced the failure, which was already present on staging HEAD.
- Related prior work: NATS satellite VRAM-sequencing (#44), NATS adapter hygiene (#49), NATS SDK adoption (#69).

--- a/src/voicecli/engine.py
+++ b/src/voicecli/engine.py
@@ -86,6 +86,11 @@ QWEN_ENGINES = frozenset({"qwen", "qwen-fast"})
 
 def get_engine(name: str) -> TTSEngine:
     engines = _get_registry()
+    if not engines:
+        raise ValueError(
+            "No engines available. Install torch for real engines, "
+            "or set VOICECLI_ENABLE_MOCK_ENGINE=1 for mock engine."
+        )
     if name not in engines:
         raise ValueError(f"Unknown engine '{name}'. Available: {list(engines.keys())}")
     return engines[name]()

--- a/src/voicecli/engine.py
+++ b/src/voicecli/engine.py
@@ -1,10 +1,13 @@
 from __future__ import annotations
 
 import contextlib
+import logging
 import re
 from abc import ABC, abstractmethod
 from collections.abc import Iterator
 from pathlib import Path
+
+log = logging.getLogger(__name__)
 
 _CUDA_PATTERNS = re.compile(
     r"CUDA|cuDNN|NCCL|out of memory|CUBLAS|CUSOLVER|GPU|"
@@ -123,8 +126,8 @@ def _get_registry() -> dict[str, type[TTSEngine]]:
                 "voxtral": VoxtralEngine,
             }
         )
-    except ImportError:
-        pass  # torch not installed — real engines unavailable
+    except ImportError as e:
+        log.debug("Skipping real engines: %s", e)
 
     if coerce_bool_env("VOICECLI_ENABLE_MOCK_ENGINE"):
         from voicecli.engines.mock import MockEngine

--- a/src/voicecli/engine.py
+++ b/src/voicecli/engine.py
@@ -98,21 +98,34 @@ def _get_registry() -> dict[str, type[TTSEngine]]:
     MockEngine is gated by ``VOICECLI_ENABLE_MOCK_ENGINE`` — unset in prod,
     set to "1" in the e2e docker-compose files so the NATS satellite can
     answer round-trip requests without loading a real model.
+
+    Real engine imports are wrapped in try/except to handle missing torch.
+    If ImportError occurs (torch not installed), real engines are skipped.
     """
-    from voicecli.engines.chatterbox import ChatterboxEngine
-    from voicecli.engines.chatterbox_turbo import ChatterboxTurboEngine
-    from voicecli.engines.qwen import QwenEngine
-    from voicecli.engines.qwen_fast import QwenFastEngine
-    from voicecli.engines.voxtral import VoxtralEngine
     from voicecli.env import coerce_bool_env
 
-    registry: dict[str, type[TTSEngine]] = {
-        "qwen": QwenEngine,
-        "qwen-fast": QwenFastEngine,
-        "chatterbox": ChatterboxEngine,
-        "chatterbox-turbo": ChatterboxTurboEngine,
-        "voxtral": VoxtralEngine,
-    }
+    registry: dict[str, type[TTSEngine]] = {}
+
+    # Try to load real engines — skip if torch unavailable
+    try:
+        from voicecli.engines.chatterbox import ChatterboxEngine
+        from voicecli.engines.chatterbox_turbo import ChatterboxTurboEngine
+        from voicecli.engines.qwen import QwenEngine
+        from voicecli.engines.qwen_fast import QwenFastEngine
+        from voicecli.engines.voxtral import VoxtralEngine
+
+        registry.update(
+            {
+                "qwen": QwenEngine,
+                "qwen-fast": QwenFastEngine,
+                "chatterbox": ChatterboxEngine,
+                "chatterbox-turbo": ChatterboxTurboEngine,
+                "voxtral": VoxtralEngine,
+            }
+        )
+    except ImportError:
+        pass  # torch not installed — real engines unavailable
+
     if coerce_bool_env("VOICECLI_ENABLE_MOCK_ENGINE"):
         from voicecli.engines.mock import MockEngine
 

--- a/tests/e2e/docker-compose.nats.yml
+++ b/tests/e2e/docker-compose.nats.yml
@@ -21,7 +21,7 @@ services:
         && uv sync --extra nats --frozen
         --no-install-package torch
         --no-install-package torchaudio -q
-        && uv run voicecli nats-serve tts --engine mock
+        && uv run --no-sync voicecli nats-serve tts --engine mock
     environment:
       VOICECLI_ENABLE_MOCK_ENGINE: "1"
       NATS_URL: "nats://nats:4222"
@@ -45,7 +45,7 @@ services:
         && uv sync --extra nats --frozen
         --no-install-package torch
         --no-install-package torchaudio -q
-        && uv run voicecli nats-serve stt --model mock
+        && uv run --no-sync voicecli nats-serve stt --model mock
     environment:
       VOICECLI_ENABLE_MOCK_ENGINE: "1"
       NATS_URL: "nats://nats:4222"

